### DISCUSSION TITLE
React Native Simulator: No bundle URL present

### DISCUSSION THREAD

**Discussion by wilsolutions** [codemagic-ci-cd/codemagic-docs#2317] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

React Native

### Steps to reproduce

1. Use the RN sample codemagic.yaml from the Code Magic repo:
https://github.com/codemagic-ci-cd/codemagic-sample-projects/blob/main/react-native/react-native-ios-simulator-build/codemagic.yaml
2. Generate the build, download the .app and install in a simulator

### Expected results

The App works without issue

### Actual results

The App display a "No bundle URL present" error:
![codemagic-react-native-simulator-build](https://github.com/codemagic-ci-cd/codemagic-docs/assets/474548/12a02de7-36c6-49ca-83f6-a41c4c75e205)

Note: this seems to be a common error when we generate a RN build without bundling it before. I believe the Code Magic sample should be updated to include the bundle command as well.

**Comment by himesh-cm** [comment]
The discussion is being continued on intercom
